### PR TITLE
Add an option to include ip-less adapters

### PR DESCRIPTION
**Pull request by memory**
Sometimes it's useful to know the MAC address of an interface
even if it does not have an IP configured

### CODE REVIEW

**Comment by memory** [comment]
Happy new year! Any thoughts on this?

**Comment by memory** [comment]
Ping? :)

**Comment by memory** [comment]
Ping? :)

**Comment by jstasiak** [comment]
Hey, seems worth having, I'll have a look if supporting this on the Win32 side is possible before merging this one (ideally the Win32 and POSIX `get_adapters()` provide the same interface.

**Comment by memory** [comment]
Added what looked to me like a reasonable pass at win32 support; note that I don't currently have a win32 device to test this on. :)

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 setup(
     name = 'ifaddr',
-    version = '0.1.6',
+    version = '0.1.7',
```
Comment (jstasiak) [inline setup.py:32]: Please leave the version intact, I'll bump it at a later point in a separate commit/PR.

File: README.rst [hunk 2]
```diff
@@ -59,6 +59,26 @@ This will print:
...
+
+    import ifaddr
+
+    adapters = ifaddr.get_adapters(include_unconfigured=True)
+
+    for adapter in adapters:
+        print "IPs of network adapter " + adapter.nice_name
+        if adapter.ips:
+            for ip in adapter.ips:
+                print "   %s/%s" % (ip.ip, ip.network_prefix)
```
Comment (jstasiak) [inline README.rst:77]: Oh, and also can you add parentheses to those print calls for Python 3 compatibility?

File: setup.py (hunk 1 above)
Comment (memory) [reply setup.py:32]: 👍

File: README.rst (hunk 2 above)
Comment (memory) [reply README.rst:77]: 👍

**Comment by jstasiak** [comment]
LGTM, I'll get back to this once I have the relevant part of the test suite done.

**Comment by jstasiak** [comment]
Cheers, it's working as expected. FYI I think I'll rename the flag from `include_unconfigured` to `include_ipless` or something, but I need to think about this in the context of another pull request.

**Comment by memory** [comment]
Woot!  And with that, my longest-lived ever PR has finally closed! 🤣✨

**Comment by jstasiak** [comment]
Contrats :)

**Pull request merged**